LiteDB/Engine/Services/WalIndexService.cs

```diff
@@ -38,7 +38,22 @@ public WalIndexService(DiskService disk, LockService locker)
         /// <summary>
         /// Get current read version for all new transactions
         /// </summary>
-        public int CurrentReadVersion => _currentReadVersion;
+        public int CurrentReadVersion
```

Comment:
Why the lock in this property? It's read-only so it shouldn't matter how many threads access it.

Reply:
The property itself is read-only, but its backing field is updatable. As I wrote in the comment above:

I made WalIndexService.CurrentReadVersion synchronized. Its backing field (_currentReadVersion) is accessed (read and modified) in the same class in Clear() and ConfirmTransaction() functions. These functions are atomic by using _indexLock. Especially ConfirmTransaction is interesting, because it first increases _currentReadVersion and then adds some data based on this increased value to the WAL index. So it seems unsafe to be able to access WalIndexService.CurrentReadVersion property while ConfirmTransaction() is still being executed.